src/modules/Thread/components/ThreadList/ThreadListItem.tsx

```diff
@@ -258,13 +259,27 @@ export default function ThreadListItem({
       )}
       {showFileViewer && (
         <FileViewer
-          message={message as FileMessage}
+          message={message as FileMessage | MultipleFilesMessage}
           isByMe={message?.sender?.userId === userId}
           onClose={() => setShowFileViewer(false)}
           onDelete={() => {
             deleteMessage(message);
             setShowFileViewer(false);
           }}
+          onDownloadClick={async (e) => {
+            if (!onBeforeDownloadFileMessage) {
+              return null;
+            }
+            try {
+              const allowDownload = await onBeforeDownloadFileMessage({ message: message as FileMessage | MultipleFilesMessage });
+              if (!allowDownload) {
+                e.preventDefault();
+                logger.info?.('ThreadListItem: Not allowed to download.');
+              }
+            } catch (err) {
+              logger.error?.('ThreadListItem: Error occurred while determining download continuation:', err);
+            }
```

Comment:
maybe we can take this repeated code out to utils and use import intead?

Reply:
yes, but let me do that later, we should merge quickly ;; 😥